core/src/main/scala/org/apache/spark/ui/scope/RDDOperationGraph.scala

```diff
@@ -164,12 +164,20 @@ private[ui] object RDDOperationGraph extends Logging {
    *
    * For the complete DOT specification, see http://www.graphviz.org/Documentation/dotguide.pdf.
    */
-  def makeDotFile(graph: RDDOperationGraph): String = {
+  def makeDotFile(graph: RDDOperationGraph, stageId: String): String = {
     val dotFile = new StringBuilder
     dotFile.append("digraph G {\n")
-    dotFile.append(makeDotSubgraph(graph.rootCluster, indent = "  "))
-    graph.edges.foreach { edge => dotFile.append(s"""  ${edge.fromId}->${edge.toId};\n""") }
-    dotFile.append("}")
+    try {
+      dotFile.append(makeDotSubgraph(graph.rootCluster, indent = "  "))
+      graph.edges.foreach { edge => dotFile.append(s"""  ${edge.fromId}->${edge.toId};\n""") }
+      dotFile.append("}")
+    } catch {
+      case t: Throwable =>
```

Comment:
Here I am not attempt to cleanup ```dotFile```, I just want to catch the Throwable to  prevent the whole page dies when the dot-file node dies.
I think one node dies can't cause the whole page dies, it will make user can't see other informations.